mlrun/datastore/sources.py

```diff
@@ -163,11 +174,7 @@ def to_spark_df(self, session, named_view=False):
 
         df = session.read.load(**self.get_spark_options())
         for col_name, col_type in df.dtypes:
-            if (
-                col_name == self.time_field
-                or self._parse_dates
-                and col_name in self._parse_dates
-            ):
+            if self._parse_dates and col_name in self._parse_dates:
```

Comment:
I might be missing something - you don't check for the `time_field` anymore, but OTOH you didn't add it to `self._parse_dates` if it existed. So, if someone still used the `time_field` it will be ignored here.

Reply:
You're right, the relevant code is in `to_step` and not the constructor. I moved it to the constructor.